compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1810,79 +1815,88 @@ private void validateOutputAttributeTypes(BLangStatement streamingQueryStatement
             }
 
-            BType structType = (((BArrayType) ((BInvokableType)
-                    ((BLangLambdaFunction) (((BLangStreamingQueryStatement) streamingQueryStatement).
-                            getStreamingAction()).getInvokableBody()).type).paramTypes.get(0)).eType);
-
-            List<BStructType.BStructField> outputStreamFieldList = ((BStructType) structType).fields;
-
-            List<? extends SelectExpressionNode> selectExpressions =
-                    ((BLangStreamingQueryStatement) streamingQueryStatement).getSelectClause().getSelectExpressions();
-
-            if (!((BLangStreamingQueryStatement) streamingQueryStatement).getSelectClause().isSelectAll()) {
-                for (int i = 0; i < selectExpressions.size(); i++) {
-                    SelectExpressionNode expressionNode = selectExpressions.get(i);
-                    BStructType.BStructField structField = null;
-                    if (expressionNode.getExpression() instanceof BLangFieldBasedAccess) {
-                        String attributeName = ((BLangFieldBasedAccess) expressionNode.getExpression()).field.value;
-                        String streamIdentifier = ((BLangSimpleVarRef) ((BLangFieldBasedAccess) expressionNode.
-                                getExpression()).expr).variableName.value;
-
-                        List<BStructType.BStructField> streamFieldList = inputStreamSpecificFieldMap.
-                                get(streamIdentifier);
-                        if (streamFieldList == null) {
-                            dlog.error(((BLangSelectClause) ((BLangStreamingQueryStatement) streamingQueryStatement).
-                                            getSelectClause()).pos, DiagnosticCode.UNDEFINED_STREAM_REFERENCE,
-                                    streamIdentifier);
-                        } else {
-                            structField = getStructField(streamFieldList, attributeName);
-                            if (structField == null) {
-                                dlog.error(((BLangSelectClause) ((BLangStreamingQueryStatement)
-                                                streamingQueryStatement).getSelectClause()).pos,
-                                        DiagnosticCode.UNDEFINED_STREAM_ATTRIBUTE, attributeName);
-                            } else {
-                                BStructType.BStructField outputStructField = outputStreamFieldList.get(i);
-                                this.types.checkType(((BLangStreamAction) ((BLangStreamingQueryStatement)
-                                                streamingQueryStatement).getStreamingAction()).pos,
-                                        outputStructField.getType(), structField.getType(),
-                                        DiagnosticCode.INCOMPATIBLE_TYPES);
+            BType streamActionArgumentType = ((BInvokableType) ((BLangLambdaFunction) (((BLangStreamingQueryStatement)
```

Comment:
Shall we please refactor this method by breaking these into smaller functions for each logical unit.

Reply:
I have moved possible logics as separate functions.